tests/cases/compiler/uncalledFunctionChecksDontWorkWithNegation.ts

```diff
@@ -0,0 +1,16 @@
+// @strictNullChecks: true
+
+declare function isFoo(): boolean;
+declare const isUndefinedFoo: (() => boolean) | undefined;
+
+if (isFoo) {
+    // error
+}
+
+if (!isFoo) {
+    // error
+}
+
+if (!isUndefinedFoo) {
+    // no error
+}
```

Comment:
would be nice to have tests for when those negated expressions are part of larger logical expressions (e.g. `!isFoo || x`, etc)